### PR TITLE
#3035 add Double data type class to avoid incorrect double(0) databas…

### PR DESCRIPTION
**Pull request by LonwoLonwo**
add Double data type class to avoid incorrect double(0) database data type result

## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

A clear and concise description of the change being made.  

As I can see, I can add a new condition to the DoubleType class, so I made these changes in the snowflake package.
Basically, this is a copy of HanaDoubleType and such kind of extension data type classes.
We fixed this problem that way in our snowflake extension fork and have used it since December. Works fine for us.

More about numeric types: https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#double-double-precision-real

I can do "DOUBLE" instead of "FLOAT", but, I think, it will be not correct.

### CODE REVIEW

**Comment by LonwoLonwo** [comment]
Hi Liquibase Team. We want to update our liquibase jat version to the latest, but I can't it properly during your last changes. Snowflake extension is now inside your main library. We use Snowflake extension fork in our project with our changes. But with the latest changes this extension do not work anymore. So, I moved our changes in your main library and created PRs:

**Comment by nvoxland** [comment]
**Initial/Pre-Review Thoughts**

Thanks for the fix, @LonwoLonwo . The changes make sense to me, I'll send it through the final check and review process to get it into a release.

Questions I have:
- None

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Comment by filipelautert** [comment]
Added a harness test at https://github.com/liquibase/liquibase-test-harness/pull/481

**Review by filipelautert — APPROVED** [review]
Change specific to ❄️ , all fine according to the documentation .

**Review by XDelphiGrl — APPROVED** [review]
This Snowflake-specific fix allows Liquibase to correctly identify the `FLOAT` datatype. This prevents generation of invalid SQL for columns defined as `DOUBLE` during `generate-changelog` and `diff-changelog`.
* Functional tests passing.
* Failing SQLServer test-harness tests unrelated to this change; not a blocker to PR merge.
  * New test harness test added for the `DOUBLE` datatype.
* No additional testing required.

APPROVED
